modules/bitcoind.nix

```diff
@@ -308,6 +308,7 @@ in {
         cfgpre=$(cat ${configFile}; printf "rpcpassword="; cat "${secretsDir}/bitcoin-rpcpassword-privileged")
         cfg=$(echo "$cfgpre" | \
         sed "s/bitcoin-HMAC-privileged/$(cat ${secretsDir}/bitcoin-HMAC-privileged)/g" | \
+        ${optionalString config.services.btcpayserver.enable ''sed "s/bitcoin-HMAC-btcpayserver/$(cat ${secretsDir}/bitcoin-HMAC-btcpayserver)/g" | \''}
```

Comment:
> Architecture-wise it would be much better to avoid any reference to btcpayserver in this file. How about we map all the passwordHMAC options to sed strings instead of manually stating them here?

Implemented ff2981e

> I don't think they're necessary really because our /secrets should be things that are useful to share across machines but this is purely an internal communication between bitcoind and btcpayserver.

I disagree, creating HMAC's locally and distributing them to your node with the correct permissions strikes me as simpler than creating a standalone service to create HMAC's.

Reply:
Simpler in some way but unnecessary in others. Okay for now.